### PR TITLE
Fix auto_pad shape inference bug

### PR DESCRIPTION
**Pull request by stevenlix**

### CODE REVIEW

File: onnx/defs/nn/defs.cc [hunk 1]
```diff
@@ -102,19 +102,19 @@ void convPoolShapeInference(
...
+          while (residual >= stride) {
+            residual -= stride;
           }
         }
-        int64_t total_pad = residual == 0 ? kernel_shape[i] - strides[i] : kernel_shape[i] + residual;
-        if (total_pad < 0) {
-          fail_shape_inference("Stride is bigger than Kernel shape");
-        }
+        int64_t total_pad = residual == 0 ? kernel_shape[i] - stride : kernel_shape[i] - residual;
+        if (total_pad < 0)
```
Comment (hariharans29) [inline onnx/defs/nn/defs.cc]: LGTM.

Just 1 naive question - do we really need this mutation ? Even if total pads is negative, in reality, this means just ignoring a few values in the raw data ("unpad" a few values) right ? Also, I think it should not affect the output dim value when we use the formula - floor((in_dim - kernel_shape + total_pad) / s) + 1. Does this seem reasonable ?

Comment (stevenlix) [reply onnx/defs/nn/defs.cc]: I checked the corner cases and I think probably there is no impact if you leave total_pad as negative value, but for simplicity and clarity, it's better to keep it as non-negative value.

Comment (stevenlix) [reply onnx/defs/nn/defs.cc]: @linkerzhang Please review this PR

**Comment by linkerzhang** [comment]
can you also add a test case to cover this bug case please? Thank you!

**Pull request merged**